src/lib/modules/job/job.go

```diff
@@ -0,0 +1,189 @@
+			Timeout: time.Second * 5,
+		},
+		request: &http.Request{},
+	}
+	if err := newReq(newJob, api); err != nil {
+		return nil, fmt.Errorf("error making http request : %w", err)
+	}
+	return newJob, nil
+}
+
+// Execute executes the monitoringJob
+func (mn *monitoringJob) Execute() {
+	for range mn.sigCh {
+		mn.JobInfo.writeTime()
+		response, err := evaluate.Monitor(mn.client, mn.request)
+		if err != nil {
+			fmt.Println(fmt.Errorf("%s", err))
+		}
+		val := fmt.Sprintf("%v|%v", response.Delay.Microseconds(), strconv.Itoa(response.Length))
+		mn.app.Append(file.NewBlock("monitoring", val))
+	}
+}
+
+// Aborts aborts the running job
+func (mn *monitoringJob) Abort() {
+	close(mn.sigCh)
+}
+
+// Info returns jobInfo of the job
+func (mn *monitoringJob) Info() *JobInfo {
+	return &mn.JobInfo
+}
+
+// newReq creates http client and request for the job
+func newReq(job *monitoringJob, api *config.API) error {
+	var body []byte
+	var err error
+	if len(api.Body) != 0 {
+		body, err = json.Marshal(api.Body)
+		if err != nil {
+			return fmt.Errorf("error marshalling : %w", err)
+		}
+	}
+	// creating http request for job
+	job.request, err = http.NewRequest(strings.ToUpper(api.Method), api.Domain+api.Route, bytes.NewBuffer(body))
+	if err != nil {
+		return fmt.Errorf("error making http request : %w", err)
+	}
+	// adding headers to the request
+	for k, v := range api.Headers {
+		job.request.Header.Set(k, v)
+	}
+	return nil
+}
+
+type machineJob struct {
+	JobInfo
+	app   file.Appendable
+	sigCh chan struct{}
+	host  string
+}
+
+func newMachineJob(app file.Appendable, c chan struct{}, api *config.API) (*machineJob, error) {
+	newjob := &machineJob{
+		app:   app,
+		sigCh: c,
+		JobInfo: JobInfo{
+			Name:        api.Name,
+			Every:       api.Every,
+			lastExecute: time.Now().Add(api.Every * -1),
+		},
+	}
+	url, err := url.Parse(api.Domain)
+	if err != nil {
+		return nil, err
+	}
+	newjob.host = url.Host
+	return newjob, nil
+}
+
+// Execute execute the machineJob
+func (mn *machineJob) Execute() {
+	for range mn.sigCh {
+		mn.JobInfo.writeTime()
+		ping, jitter, err := evaluate.Machine(mn.host)
+		if err != nil {
+			fmt.Println(fmt.Errorf("%s", err))
+		}
+		pingval := fmt.Sprintf("%v|%v|%v", ping.Max.Microseconds(), ping.Mean.Microseconds(), ping.Mean.Microseconds())
+		jitterval := fmt.Sprintf("%v", jitter.Value.Microseconds())
+		mn.app.Append(file.NewBlock("ping", pingval))
+		mn.app.Append(file.NewBlock("jitter", jitterval))
+	}
+}
+
+// Abort aborts the running job
+func (mn *machineJob) Abort() {
+	close(mn.sigCh)
+}
+
+// Info returns the jobInfo of the job
+func (mn *machineJob) Info() *JobInfo {
+	return &mn.JobInfo
+}
+
+// ReadTime reads lastExecute of the JobInfo
+func (j *JobInfo) ReadTime() time.Time {
+	j.mux.RLock()
+	t := j.lastExecute
+	j.mux.RUnlock()
+	return t
+}
+
+// writeTime writes lastExecute of the JobInfo
+func (j *JobInfo) writeTime() {
+	j.mux.Lock()
+	j.lastExecute = time.Now()
+	j.mux.Unlock()
+}
```

Comment:
These functions should be just after where the `JobInfo` struct is declared for better readability.